src/pages/PrimeCalculator/Form/index.tsx

```diff
@@ -14,75 +20,137 @@ import {
   Spinner,
   TokenIconWithSymbol,
 } from 'components';
+import FunctionKey from 'constants/functionKey';
+import useDebounceValue from 'hooks/useDebounceValue';
 import { useGetToken } from 'packages/tokens';
 import { useTranslation } from 'packages/translations';
-import { Token } from 'types';
-import { areAddressesEqual } from 'utilities';
+import { useAccountAddress, useChainId } from 'packages/wallet';
+import { Asset } from 'types';
+import {
+  areAddressesEqual,
+  convertMantissaToTokens,
+  formatCentsToReadableValue,
+  formatTokensToReadableValue,
+} from 'utilities';
 
 import TEST_IDS from '../testIds';
 import { Field } from './Field';
 import { RewardDetails } from './RewardDetails';
+import { getInitialValues } from './getInitialValues';
 
 export const QUERY_PARAM_TOKEN_ADDRESS = 'tokenAddress';
 
-const formSchema = z.object({
-  stakedAmountXvs: z.string().min(1),
-  suppliedAmountTokens: z.string().min(1),
-  borrowedAmountTokens: z.string().min(1),
-});
+const validateNumericString = (str: string, minValue = 0) => {
```

Comment:
Should this be organized under utilities?

Reply:
Moved to its own file, just not under utilities, since it wouldn't be imported in other components